work13/index.js

```diff
@@ -0,0 +1,135 @@
+"use strict";
+const wrap = document.getElementById("js_wrap");
+const ul = document.getElementById("js_list");
+const modalBtn = document.getElementById("js_modalButton");
+const modal = document.getElementById("js_modal");
+
+const url = "https://mocki.io/v1/d6da0b8a-3546-419e-aaec-05f3a247c6d0";
+// const url = "https://myjson.dit.upm.es/api/bins/ほげほげajy3";
+// const url = "https://myjson.dit.upm.es/api/bins/eu2f";
+
+// 下記は、myjson繋がらない時の固定値
+// const url = {
+//   "data": [
+//     {
+//       "a": "bookmark",
+//       "img": "img/1.png",
+//       "alt": "画像１",
+//       "text": "ブックマーク"
+//     },
+//     {
+//       "a": "message",
+//       "img": "img/2.png",
+//       "alt": "画像２",
+//       "text": "メッセージ"
+//     }
+//   ]
+// }
+
+async function getData() {
+  try {
+    const response = await fetch(url);
+    if (response.ok) {
+      const json = await response.json();
+      return json;
+    } else {
+      // .okでなければerrorを投げる
+      throw new Error(`Server request failed:${response.statusText}`);
+    }
+    // 下記は、固定値をそのままpromiseの返り値とする
+    // const json = url
+    // return json
+  } catch (e) {
+    //上記のthrowでエラーが出ていれば、ここでコンソールに表示し、
+    //エラーを投げる
+    console.error(e)
+    throw new Error(e);
+  }
+}
+
+async function getListData() {
+  let listData;
+  try {
+    listData = await getData();
+  } catch (e) {
+    wrap.textContent = `エラー内容:${e.message}`;
+  } finally {
+    removeLoading();
+  }
+  if (listData.data.length === 0) {
+    wrap.textContent = "data is empty";
+    return;
+  }
+  return listData;
+}
+
+function removeLoading() {
+  const loading = document.querySelector(".loading")
```

Comment:
IMO
I think it is safe to give the loading an id and use getElementId.

Reply:
I investigated the difference.
"querySelector ()" is slower than "getElementById ()", isn't it?
Those that can use "id" use "getElementById ()".

commit:

Reply:
nits
I don't think you have to define it in a variable.
Because it is not used elsewhere.
```js
  document.getElementById("js-loading").remove();
```

Reply:
I was trying to put it in a variable every time.
I was just lengthening the code ...

thank you
commit: [b73d31d](https://github.com/yasu-pro/js-handson/pull/15/commits/b73d31d74cb2efffd0367fe36e5e051c3c1d7e6c)